### DISCUSSION TITLE
[handlebars] String related helper functions

### DISCUSSION THREAD

**Discussion by mhanberg** [zk-org/zk#61] [Ideas]
Hey!

I am setting up a group and command alias for my one on one/syncs at work, and I've reach a point where I think some additional helper functions in the handlebars syntax would be nice.

I have a command alias set up called `sync = "zk new --group=sync --extra with=$@"`, that I call as `zk sync bob`.

This fills in the following template

```handlebars
---
title: "{{ extra.with }} - {{ date now "full" }} - {{ date now "time" }}"
tags: [sync, {{ extra.with }}]
date: {{ date now "full" }} - {{ date now "time" }}
---

{{content}}
```

I would like to be able to `downcase` the tag and `capitalize` the section of the title, so that I can use the alias as `zk sync bob` or `zk sync Bob` and not have to worry about inconsistencies in the title and having duplicated tags. (Another thought is that tags could be case insensitive, but that is unrelated).

I think I can probably get by with some shell functions for now, but let me know what you think.

Thanks!

**Comment by mickael-menu** [comment]
I'm not against adding some string helpers in the Handlebars template. I actually added a few like `{{concat}}`. I'm just a bit afraid of ending up maintaining a whole toolkit in the template. 😄 

An alternative would be to directly use shell commands in the template, with `{{sh}}`. Passing arguments to the shell command is not straightforward right now, but you could handle your example like this:

```
---
title: "{{ extra.with }} - {{ date now "full" }} - {{ date now "time" }}"
tags: [sync, {{#sh "tr '[:upper:]' '[:lower:]'"}}{{ extra.with }}{{/sh}}]
date: {{ date now "full" }} - {{ date now "time" }}
---

{{content}}
```

Maybe the `{{sh}}` helper could be improved to pass arguments, to be able to write directly `{{sh "tr '[:upper:]' '[:lower:]'" extra.with}}`.

**Reply by mhanberg** [reply]
I ended up with two zsh scripts that i call from within the alias

```zsh
#!/usr/bin/env zsh
# Capitalize
echo "${(C)1}"
```
```zsh
#!/usr/bin/env zsh
# Downcase
echo "${(L)1}"
```

```toml
sync = "zk new --group=sync --extra with=$(~/.bin/capitalize $1),tag=$(~/.bin/downcase $1)"
```

**Reply by mickael-menu** [reply]
That works too :) 

You could even use directly the `zsh` expansion in the alias. `zk` uses `$SHELL -c ...` to expand aliases.

```toml
sync = "zk new --group=sync --extra with=${(C)1},tag=${(L)1}"
```
